app/src/main/java/com/google/droidjump/leveldata/LevelGenerator.java

```diff
@@ -0,0 +1,47 @@
+/*
+ * Copyright 2020 Google LLC
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     https://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.google.droidjump.leveldata;
+
+import static com.google.droidjump.leveldata.ObstacleType.BAT;
+import static com.google.droidjump.leveldata.ObstacleType.CACTUS;
+import static com.google.droidjump.leveldata.ObstacleType.PALM;
+import java.util.Random;
+
+/**
+ * Generates obstacles.
+ * In later improvements could base generation on custom parameters.
+ */
+public class LevelGenerator {
+    static private final int MIN_INTERVAL_VALUE = 20; // Min value of interval between obstacles.
+    static private final int MAX_INTERVAL_VALUE = 80; // Max value of interval between obstacles.
+    // Options of obstacles for generation.
+    static private final ObstacleType[] OBSTACLE_OPTIONS = {CACTUS, CACTUS, CACTUS, PALM, PALM, BAT};
```

Comment:
Do I understand correctly that you put several instances to make cactus more frequently than palm and palm more frequently than bat?

I guess this is an OK way to do it now but it might not scale well if we decide that proportion should be different or make it configurable. Do you have an idea how to do it?

We can store a prefix sum of these proportions, for example. If we want chances to be 100:73:7, we would store [100, 173, 180] and compare the generated random value with them to determine which obstacle to create. If the number of obstacle types grows, we would use binary search to find the correct obstacle type (no need for it now though). What do you think?

Reply:
Yes, I thought about it. I just implemented the simplest idea I had. The idea with prefix sums sounds good. Moreover, these may not necessarily be sums. We can directly set the proportions, nothing really changes from this. Ok, I really think it's better to implement in a more scalable way. I will pass proportions as parameters of method.